specification/digitaltwins/data-plane/Microsoft.DigitalTwins/preview/2020-05-31-preview/digitaltwins.json

```diff
@@ -0,0 +1,1497 @@
+            }
+          },
+          "default": {
+            "description": "Default response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/digitaltwins/{id}": {
+      "get": {
+        "tags": [
+          "Twins"
+        ],
+        "operationId": "DigitalTwins_GetById",
+        "description": "Retrieves a digital twin.\nStatus codes:\n200 (OK): Success.\n404 (Not Found): There is no digital twin with the provided id.",
+        "x-ms-examples": {
+          "GetTwin": {
+            "$ref": "./examples/GetTwin.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/digitalTwinId"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "type": "object"
+            },
+            "headers": {
+              "ETag": {
+                "description": "Weak Etag.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Default response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Twins"
+        ],
+        "operationId": "DigitalTwins_Add",
+        "description": "Adds or replaces a digital twin.\nStatus codes:\n200 (OK): Success.\n400 (Bad Request): The request is invalid.\n412 (Precondition Failed): The model is decommissioned or the digital twin already exists (when using If-None-Match: *).",
+        "x-ms-examples": {
+          "PutTwinBasicExample": {
+            "$ref": "./examples/PutTwinBasicExample.json"
+          },
+          "PutTwinAdvancedExample": {
+            "$ref": "./examples/PutTwinAdvancedExample.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/digitalTwinId"
+          },
+          {
+            "name": "twin",
+            "description": "The digital twin instance being added. If provided, the $dtId property is ignored.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "type": "object"
+            }
+          },
+          {
+            "$ref": "#/parameters/if-none-match-star"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "type": "object"
+            },
+            "headers": {
+              "ETag": {
+                "description": "Weak Etag.",
+                "type": "string"
+              }
+            }
+          },
+          "202": {
+            "description": "Asynchronous Success"
```

Comment:
Does this mean it's an async operation (i.e. long running operation)?  If so then this operation should be annotated with `x-ms-long-running-operation`.  Please see https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-long-running-operation for details.

Reply:
No, it's not a LRO in the x-ms-long-running-operation sense. This API can return 200 when the digital twin entity is created/updated immediately or 202 when the entity is created/updated and requires a device sync. In that case, the client uses the digital twin notification feed to get notifications about entity lifecycle and state changes.